instructor/anthropic_utils.py

```diff
@@ -0,0 +1,65 @@
+import re
+import json
+import xmltodict
+from pydantic import BaseModel
+import xml.etree.ElementTree as ET
+from typing import Type, Any, Dict, TypeVar
+
+T = TypeVar('T', bound=BaseModel)
+
+def json_to_xml(json_str: str) -> str:
+    """Takes a JSON string representing a Pydantic model and returns XML format for Anthropic function calling."""
+    try:
+        model_dict = json.loads(json_str)
+    except json.JSONDecodeError:
+        raise ValueError("Invalid JSON string.")
+    
+    if "properties" not in model_dict:
+        raise ValueError("JSON structure is not as expected.")
+
+    root = ET.Element("tool_description")
+    tool_name = ET.SubElement(root, "tool_name")
+    tool_name.text = model_dict.get("title", "Unknown")
+    description = ET.SubElement(root, "description")
+    description.text = "This is the function that must be used to construct the response."
+    parameters = ET.SubElement(root, "parameters")
+    
+    references = model_dict["$defs"] if "$defs" in model_dict else None # Pass in nested structures if they exist
+    _add_params(parameters, model_dict, references)
+
+    return ET.tostring(root, encoding="unicode")
+
+def _add_params(root: ET.Element, model_dict: Dict[str, Any], references: Dict[str, Any]) -> None: # check to handle nested params with the same name?
+    properties = model_dict.get("properties", {})
+            
+    for field_name, details in properties.items():
+        parameter = ET.SubElement(root, "parameter")
+        name = ET.SubElement(parameter, "name")
+        name.text = field_name
+        type_element = ET.SubElement(parameter, "type")
+        type_element.text = details.get("type", "unknown") # Might be better to fail here if there is no type since pydantic models require types
+        param_description = ET.SubElement(parameter, "description")
+        param_description.text = f"The {field_name} of the {model_dict['title']} model"
```

Comment:
shuoldn't this be part of the model, not the generated str we produce for them?

Reply:
another way to explain this is that theres literally an attribute called description we can use rather than making up `f"The {field_name} of the {model_dict['title']} model"`